Sources/_StringProcessing/Utility/TypeConstruction.swift

```diff
@@ -99,6 +99,13 @@ public func tupleType<ElementTypes: BidirectionalCollection>(
   }
 }
 
+// FIXME: namespaces are hard
+internal func _tuple<Elements: BidirectionalCollection>(
+  of elements: __owned Elements
+) -> Any where Elements.Element == Any {
+  tuple(of: elements)
+}
+
```

Comment:
Any reason we need an additional `_tuple(of:)`, instead of using `tuple(of:)` defined below?

Reply:
The call site originally stated `Util.tuple` because it was ambiguous and/or shadowed by `Capture.tuple`. Now that it's the same module, we don't have that disambiguation option. It might not be necessary here, I had tried doing this change when the module was still named `Regex` which was particularly unhelpful due to `Regex<Capture>`.

Reply:
```
StringProcessing/Sources/_StringProcessing/Capture.swift:26:33: error: cannot convert value of type '[Any]' to expected argument type '[Capture]'
      return tuple(of: elements.map(\.value))
                                ^
```

Reply:
Also, `tupleType(of:)` and `tuple(of:)` should be made internal since they aren't user-facing.

Reply:
> The call site originally stated Util.tuple because it was ambiguous and/or shadowed by Capture.tuple. Now that it's the same module, we don't have that disambiguation option.

Maybe move `tuple(of:)` and `tupleType(of:)` as static methods under an empty enum `TypeConstruction`?

Reply:
Re. error, we should make `Capture` internal.

Reply:
Done